sdk/storage/azure-storage-common/test/ut/test_base.hpp

```diff
@@ -0,0 +1,211 @@
+          const std::string& queryParameters);
+
+      /* cspell:disable-next-line */
+      constexpr static const char* DummyMd5 = "tQbD1aMPeB+LiPffUwFQJQ==";
+      /* cspell:disable-next-line */
+      constexpr static const char* DummyCrc64 = "+DNR5PON4EM=";
+
+      uint64_t RandomInt(
+          uint64_t minNumber = std::numeric_limits<uint64_t>::min(),
+          uint64_t maxNumber = std::numeric_limits<uint64_t>::max());
+
+      std::string RandomString(size_t size = 10);
+
+      std::string GetStringOfSize(size_t size = 10, bool lowercase = false);
+
+      std::string LowercaseRandomString(size_t size = 10);
+
+      Storage::Metadata GetMetadata(size_t size = 5);
+
+      void RandomBuffer(char* buffer, size_t length);
+      inline void RandomBuffer(uint8_t* buffer, size_t length)
+      {
+        RandomBuffer(reinterpret_cast<char*>(buffer), length);
+      }
+      std::vector<uint8_t> RandomBuffer(size_t length);
+
+      inline std::vector<uint8_t> ReadBodyStream(
+          std::unique_ptr<Azure::Core::IO::BodyStream>& stream)
+      {
+        Azure::Core::Context context;
+        return stream->ReadToEnd(context);
+      }
+
+      inline std::vector<uint8_t> ReadBodyStream(
+          std::unique_ptr<Azure::Core::IO::BodyStream>&& stream)
+      {
+        return ReadBodyStream(stream);
+      }
+
+      std::vector<uint8_t> ReadFile(const std::string& filename);
+
+      void DeleteFile(const std::string& filename);
+
+      std::string InferSecondaryUrl(const std::string primaryUri);
+
+      inline std::string Base64EncodeText(const std::string& text)
+      {
+        return Azure::Core::Convert::Base64Encode(std::vector<uint8_t>(text.begin(), text.end()));
+      }
+
+      virtual void SetUp() override
+      {
+        Azure::Core::Test::TestBase::SetUpTestBase(AZURE_TEST_RECORDING_DIR);
+      }
+
+    public:
+      const static Azure::ETag DummyETag;
+      const static Azure::ETag DummyETag2;
+    };
+
+    constexpr inline unsigned long long operator""_KB(unsigned long long x) { return x * 1024; }
+    constexpr inline unsigned long long operator""_MB(unsigned long long x)
+    {
+      return x * 1024 * 1024;
+    }
+    constexpr inline unsigned long long operator""_GB(unsigned long long x)
+    {
+      return x * 1024 * 1024 * 1024;
+    }
+    constexpr inline unsigned long long operator""_TB(unsigned long long x)
+    {
+      return x * 1024 * 1024 * 1024 * 1024;
+    }
+
+    class CryptFunctionsTest : public StorageTest {
+    };
+
+    class ClientSecretCredentialTest : public StorageTest {
+
+    private:
+      std::unique_ptr<Azure::Storage::Blobs::BlobContainerClient> m_client;
+
+    protected:
+      std::shared_ptr<Core::Credentials::TokenCredential> m_credential;
+      std::string m_containerName;
+
+      // Required to rename the test propertly once the test is started.
+      // We can only know the test instance name until the test instance is run.
```

Comment:
It looks to me `testName` is just test suite name + test case name. Can you help me understand why we need to wait until the test case is run and do the renaming?